### ISSUE TITLE
Apple Silicon client support

### ISSUE THREAD

**Issue by air720boarder** [rustdesk/rustdesk#368]
It would be great for you to add in Apple Silicon support to your Mac app to make it universal!

**Comment by realitymolder** [comment]
True, very important for native support and much greater performance.

**Comment by Evgir** [comment]
I could build own version of Apple Silicon native, simply change the magnum-opus git in the Cargo.toml file to https://github.com/clovistx/magnum-opus

**Comment by rustdesk** [comment]
will be in 1.2

**Comment by leland37** [comment]
You say this will be in 1.2; does that mean you are testing it in the nightly currently?

**Comment by luckman212** [comment]
Is there an Arm (AppleSilicon) build of 1.2.0 now that the release is out?

**Comment by wavedeck** [comment]
Release 1.2.0 is out now, but the build for apple silicon is still missing :/

**Comment by RahulSajnani** [comment]
Hi! Is this available now? I don't see any pre-built binary for m2 macs.

**Comment by dinger1986** [comment]
having access to one would help the developers greatly, if anyone wants to help contribute?

**Comment by wavedeck** [comment]
I might give it a try to compile a binary on my (18,3) M1 MacBook Pro to see if it’s even possible without dependency / platform issues. Let’s see if it goes well. ^^ 

Other than that i would give Scaleway a try (a cloud compute as a service) if the project is actually interested in providing binaries for apple silicon. One could easily plug it into CI/CD for building binaries on cloud provided M1 Mac Minis. 

This also helps with trust, as compiled binaries from community members can’t be guaranteed to not be injected with malware.

**Comment by wavedeck** [comment]
Using the [macOS build docs](https://rustdesk.com/docs/en/dev/build/osx/) for rustdesk client i am able to compile a debug build and run rustdesk as a native apple silicon binary. That went relatively smooth so far 🥳

Unfortunately, trying to create an optimized build using `cargo build --release` fails due to missing header files for `opus/opus_multistream.h`. Maybe this is due to having opus already installed via brew and i didn't bother to also install it via vspkg too. 

TLDR;
Although it seems possible to compile the current stable release `v1.2.1` on apple silicon, I wouldn't advise for anyone to build a debug image and using it in production. Instead, run the x86 image using rosetta instead. I believe the performance benefits are negligible anyways. (Further tests and a successful release build have to be made before this can be validated)

I'll keep you updated if I'm able to resolve the dependency issue with opus and successfully build an optimized image on my Mac
